code/components/ros-patches-five/src/AccountID.cpp

```diff
@@ -97,7 +98,7 @@ struct ExternalROSBlob
 	uint8_t data[16384];
 	uint8_t steamData[64 * 1024];
 	size_t steamSize;
-	uint8_t epicData[64 * 1024];
+	char epicData[64 * 1024];
```

Comment:
Why not go for `std::string` instead? Looking at the code below, special functions are now put in place to keep everything working and null-terminated, truncation on `StringCbCopyA` and `StringCbCatA` also doesn't seem desirable.

Reply:
This structure is shared cross-process via HostSharedData, so has to be fixed-size and not contain any external pointers. `std::string` would be awfully unsuited for this.

The case where the command-line string will be truncated in and of itself is extremely unlikely since we launch this subprocess ourselves, the entire request itself was solely to meet the 'solve the automated security scan' bullet point.

Additionally, the 65535 length is because the process command line [can't exceed this length](https://devblogs.microsoft.com/oldnewthing/20031210-00/?p=41553). It can't possibly get truncated.

Reply:
I see, thank you. Making comments in code on the last points should be wise then, don't need to look up this PR again :)

Reply:
> extremely unlikely

Or impossible? If the former, some window-dressing/defensiveness (informal: log spam) could be added to `GetCommandLineAStub` to at least handle those silently routines failing. It would also provide feedback on:

> truncation on StringCbCopyA and StringCbCatA also doesn't seem desirable.